code/game/gamemodes/objective.dm

```diff
@@ -517,16 +587,19 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	martyr_compatible = 1
 	var/target_real_name
 
-/datum/objective/destroy/find_target()
-	var/list/possible_targets = active_ais(1)
-	var/mob/living/silicon/ai/target_ai = pick(possible_targets)
-	target = target_ai.mind
+/datum/objective/destroy/set_explaination_text()
 	if(target && target.current)
 		target_real_name = target.current.real_name
 		explanation_text = "Destroy [target_real_name], the AI."
 	else
-		explanation_text = "Free Objective"
-	return target
+		..()
+
+/datum/objective/destroy/find_target()
+	var/list/possible_targets = active_ais()
+	var/mob/living/silicon/ai/target_ai = safepick(possible_targets)
+	var/found_target = target_ai?.mind
+	set_target(found_target)
+	return found_target
```

Comment:
Similar to a bit above, seems like it might be better to just return early if there's no active AIs rather than putting failsafes in.
```suggestion
/datum/objective/destroy/find_target()
	var/list/possible_targets = active_ais()
	if(!length(possible_targets))
		return null
	var/found_target = pick(possible_targets)
	set_target(found_target)
	return found_target
```

Reply:
`find_target` is always supposed to set the `target` value. Even if this becomes `null`. I'll change this in the other instances as well to reflect this better.